### PR TITLE
first draft for vega plot blog

### PR DESCRIPTION
**Pull request by jspaaks**
- added cli;
- added vector-based public 'iterations' member to NewtonRaphson class;
- changed the indent to 2

Feel free to Undraft and/or work on this -- I won't be working on this the next few days.

### CODE REVIEW

**Comment by sverhoeven** [comment]
Move code to js-webapp-plot/ directory.

**Comment by sverhoeven** [comment]
Dropped cli as the iterations can be seen in console log in JS.

File: js-plot/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Show me the visualization
+
+Running C++ code in a web browser is all nice, but we really want to grab someones attention by visualizing something. In this blog we are going to make a plot from the results coming from our C++ code.
+
+To make a plot we needs some data. In a [previous](../js-webapp) story we found the root of an equation using the NewtonRaphson algorithm implemented in C++ and compiled to a WebAsssembly module.
```
Comment (fdiblen) [inline js-plot/README.md]:
```suggestion
To make a plot we need some data. In a [previous](../js-webapp) post we found the root of an equation using the Newton-Raphson algorithm implemented in C++ and compiled to a WebAsssembly module.
```

File: js-plot/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+A single root value makes for a depressing plot. The NewtonRaphson algorithm uses iterations to find the root. So we will capture the data of each iteration and plot those.
```
Comment (fdiblen) [inline js-plot/README.md]:
```suggestion
A single root value makes for a depressing plot. The Newton-Raphson algorithm uses iterations to find the root so we will capture the data of each iteration and plot those.
```

File: js-plot/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Let make changes to the C++ code to retrieve the data from all of the iterations.
```
Comment (fdiblen) [inline js-plot/README.md]:
```suggestion
Let's make changes to the C++ code to retrieve the data from the iterations.
```

File: js-plot/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+TODO add code snippets
+
+TODO show iterations from intiaizas json doc
+
+TODO Introduce Vega
+
+TODO explain vega-spec.json layer construction
+
+When we open the page we will be greeted by the following plot. We can zoom with scrollwheel and pan by dragging.
```
Comment (fdiblen) [inline js-plot/README.md]:
```suggestion
When we visit the web page, we will be greeted by the following plot. We can zoom with mouse wheel and pan by dragging.
```

File: js-plot/index.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  </head>
+  <body>
+    <div id="plot"></div>
+    <script>
+      createModule().then(async (rootfinding) => {
+        const tolerance = 0.001;
+        const initial_guess = -4;
+        const newtonraphson = new rootfinding.NewtonRaphson(tolerance);
+        newtonraphson.solve(initial_guess);
+        const iterations = new Array(
```
Comment (fdiblen) [inline js-plot/index.html:16]: We need to explain what is happening between line 16 and 31. It won't be obvious for the readers that are not familiar with JavaScript.

File: js-plot/vega-spec.json [hunk 6]
```diff
@@ -0,0 +1,100 @@
+{
```
Comment (fdiblen) [inline js-plot/vega-spec.json:1]: Do we still want log scale here? If user has a big initial guess, we will again have a problem with the plot.

**Review by fdiblen — CHANGES_REQUESTED** [review]
Fancy fancy...
:trophy: 

I have suggested a few changes.

File: js-plot/vega-spec.json (hunk 6 above)
Comment (sverhoeven) [reply js-plot/vega-spec.json:1]: Could not get log scale to work together with layers and zoom/pan.

Initial guess is still hardcoded, so if the user changes it he/she should also change the start+stop .

File: js-plot/index.html (hunk 5 above)
Comment (sverhoeven) [reply js-plot/index.html:16]: Indeed, added todo in 4162e1e7d370490dcb3f5325e26d1a326f9eb1e2

**Review by fdiblen — APPROVED** [review]
:100:

**Pull request merged**